src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -656,10 +656,51 @@ def update_period(self):
             exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd)
             if exit_code > 0:
                 raise DashboardException(e=err, msg='Unable to update period',
-                                         http_status_code=500, component='rgw')
+                                         http_status_code=400, component='rgw')
         except SubprocessError as error:
             raise DashboardException(error, http_status_code=500, component='rgw')
 
+    def edit_realm(self, realm_name: str, default: bool, new_realm_name: str):
+        rgw_realm_edit_cmd = []
+        cmd_edit_realm_options = []
+        if new_realm_name == realm_name:
+            if str_to_bool(default):
```

Comment:
Why is `default: bool` a boolean and then being transformed to a string? Either use a correct type for the parameter or pass the argument `default` correctly.

Reply:
Similar to this https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/controllers/host.py#L310

Reply:
Then 
```
    def edit_realm(self, realm_name: str, default: bool, new_realm_name: str):
should

Reply:
But don't remove types to solve the issue :cry:

Reply:
Well the format for default param changes. PYthon will understand bool in form of `True` or `False` , whereas from frontend we pass it as `true`/`false` which isn;t bool for python. So it takes input in form of string & thus we need `str_to_bool`. It's the same scenario in other components as well, like in case og host list which I mentioned before.

Reply:
That means we can establish a contract in the endpoint to adhere to a real boolean instead of sending a string in the frontend as I pointed out in hosts.set, 100 lines below the list. My point being, we should try to use type annotations as much as possible, and correctly, to deal with the python's macabre type system.

Reply:
The main issue I see with this is due to the use of `HttpParams`, as it transforms any given value to a string.